coderd/userauth.go

```diff
@@ -636,19 +877,27 @@ func (api *API) oauthLogin(r *http.Request, params oauthLoginParams) (*http.Cook
 		return nil
 	}, nil)
 	if err != nil {
-		return nil, xerrors.Errorf("in tx: %w", err)
+		return nil, database.APIKey{}, xerrors.Errorf("in tx: %w", err)
 	}
 
-	cookie, err := api.createAPIKey(ctx, createAPIKeyParams{
+	cookie, key, err := api.createAPIKey(ctx, createAPIKeyParams{
 		UserID:     user.ID,
 		LoginType:  params.LoginType,
 		RemoteAddr: r.RemoteAddr,
 	})
 	if err != nil {
-		return nil, xerrors.Errorf("create API key: %w", err)
+		return nil, database.APIKey{}, xerrors.Errorf("create API key: %w", err)
 	}
 
-	return cookie, nil
+	return cookie, *key, nil
+}
+
+// setAnonUser passes a disposable user ID to force an audit diff
+// and generate a log for a failed login
+func setAnonUser(request *audit.Request[database.APIKey]) {
```

Comment:
Just provide more context. How/when it is appropriate to use the method?
```suggestion
// setAnonUser should only be used on unauthenticated routes where the user
// making the request is unknown. To force audit logs to show an unknown user
// made the request, a disposable user ID is used.
// This user id has no relation to any actual user and indicates
// a failed attempt (eg failed login) of an anonymous user.
func setAnonUser(request *audit.Request[database.APIKey]) {
```